### ISSUE TITLE
Possible awareness of Shader Patch's tools?

### ISSUE THREAD

**Issue by SleepKiller** [Gametoast/ZeroMunge#55]
I recently tried to use this with a world that was using SP's tools for custom materials. When I did I had this pop up.

![zeromunge_load_capture](https://user-images.githubusercontent.com/5582558/54507984-ca5df780-49a8-11e9-9962-b5ece43d2e72.png)

As I'm unfamiliar with the Zero Munge's inner workings so it'd be useless for me to speculate as to what causes this. (Or what the exact consequences of it are.) But I can provide details on the edits to `munge_world.bat` that Shader Patch requires.

First below the invocation of the game's regular mungers (but before `levelpack` is invoked) these lines are put.
```batch
@REM Shader Patch Mungers
set MUNGE_DIR_SP=%MUNGE_DIR%_SP
set OUTPUT_DIR_SP=%OUTPUT_DIR%\SP

material_munge --sourcedir %SOURCE_DIR% --mungedsourcedir %MUNGE_DIR% --outputdir %MUNGE_DIR_SP% --descdir %MUNGE_BIN_DIR%\rendertype_descriptions 2>>%MUNGE_LOG%
sp_texture_munge --sourcedir %SOURCE_DIR% --outputdir %MUNGE_DIR_SP% 2>>%MUNGE_LOG%

@REM Shader Patch Effects Handling
spfx_munge --sourcedir %SOURCE_DIR% --outputdir %MUNGE_DIR_SP% 2>>%MUNGE_LOG%
configmunge -inputfile *.tmpfx -sourcedir %MUNGE_DIR_SP% -outputdir %MUNGE_DIR_SP% -chunkid fx -ext game_envfx 2>>%MUNGE_LOG%
```

And then at the end of the loop for packing the world's lvls these two commands are placed.
```batch
@REM Shader Patch uses it's own version of `levelpack` that provides strict guarantees for the search order of input folders.
lvl_pack --sourcedir %%a --outputdir %MUNGE_DIR_SP% -f ".+\.mrq" -i %MUNGE_DIR_SP% -i %MUNGE_DIR% -i %WORLDS_COMMON_MUNGE_DIR% -e %COMMON_MUNGE_DIR%\core.files -e %COMMON_MUNGE_DIR%\common.files -e %COMMON_MUNGE_DIR%\ingame.files 2>>%MUNGE_LOG%
lvl_pack --sourcedir %%a --outputdir %OUTPUT_DIR_SP% -i %MUNGE_DIR_SP% -i %MUNGE_DIR% -i %WORLDS_COMMON_MUNGE_DIR% -e %COMMON_MUNGE_DIR%\core.files -e %COMMON_MUNGE_DIR%\common.files -e %COMMON_MUNGE_DIR%\ingame.files 2>>%MUNGE_LOG%
```

`munge_side.bat` has slightly different integration (namely the packing of child lvls and the commands under Effects Handling are not present.) Let me know if you'd like more details.

I'd hate for people that use Zero Munge to be unable to use Shader Patch's tools or vice versa. I'm unsure on what exactly needs to be changed in either of them to make their usage together be seamless but I'd definitely like to work towards making it so.

**Comment by marth8880** [comment]
Will have a look at this for the next release!

**Comment by marth8880** [comment]
Fixed in 7a6f0873b642f7c37f3933c226231a37400f0f4f
